packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -109,6 +118,8 @@ class ThreadChannel extends Channel {
       this.invitable ??= null;
     }
 
+    this.createdTimestamp ??= this.type === ChannelType.GuildPrivateThread ? super.createdTimestamp : null;
```

Comment:
How is this true and how would super.createdTimestamp be defined if the one here isn’t?

Reply:
super would take from the base class 👀

Reply:
Yeah I know but how would that work here?

Reply:
> Yeah I know but how would that work here?

If a thread has a starter message the tread id will be the starter message id, else it will be unique id. We can be sure that the id is unique only when the channel type is GuildPrivateThread. So we can be sure that createdTimestamp is correct only when the channel type is GuildPrivateThread.